lib/fog/openstack/core.rb

```diff
@@ -119,6 +116,56 @@ def set_api_path
         # if the service supports multiple versions, do the selection here
       end
 
+      def set_microversion
+        # both need to be set in service's initialize for microversions to work
+        return if !@supported_microversion || !@supported_versions
+
+        @microversion_key          ||= 'Openstack-API-Version'.freeze
+        @microversion_service_type ||= @openstack_service_type.first
```

Comment:
at line https://github.com/sapcc/fog-openstack/blob/bce4837ac29502affd4d5ea6c5be00cb2378756e/lib/fog/openstack/core.rb#L154

for how [compute](http://developer.openstack.org/api-guide/compute/microversions.html#client-interaction)/[identity](https://specs.openstack.org/openstack/api-wg/guidelines/microversion_specification.html#client-interaction) use the microversion with 'Openstack-API-Version' Header in contrast to [shared file system](http://docs.openstack.org/developer/manila/devref/api_microversion_dev.html), where we don't have that (yet).
And I need to be able to specify it for each service, because it can differ from the service_type (f.i. 'compute' vs. 'computev2')